### PR TITLE
(Docs) Improve troubleshooting doc

### PR DESCRIPTION
**Pull request by hestonhoffman**
Moved the bit on tempdirs into its own section so it's easier
for people to scan and find the info they're looking for

!no-release-note

### CODE REVIEW

File: documentation/troubleshooting.md [hunk 1]
```diff
@@ -26,15 +26,7 @@ You need to upgrade your project so Bolt can manage your modules and
...
-## My task fails mysteriously
-
-Try running Bolt with `--log-level debug` to see the exact output from your task.
-
-Make sure your task executable starts with a shebang (`#!`) line indicating the
-interpreter to use and verify that the executable is present on the target
-system. For example, if you write a Python task and include the line:
-`#!/usr/bin/env python`, Bolt attempts to execute the script using the default
-`python` executable on the target system.
+## My task fails with a "permission denied" error
```
Comment (lucywyman) [inline documentation/troubleshooting.md:29]: It sounds like Dan might have been specifically looking for `noexec` here...I'm not sure how to phrase this such that it includes that? Something like `## My task fails with a "permission denied" error - could be that tmpdir is noexec` but...obviously shorter :grimacing:

Comment (hestonhoffman) [reply documentation/troubleshooting.md:29]: Doesn't that mean you'd be looking for the problem using the solution?

Comment (hestonhoffman) [reply documentation/troubleshooting.md:29]: Okay, I think I understand the issue. Let me take another stab at it

Comment (beechtom) [reply documentation/troubleshooting.md:29]: Considering we've had several users run into this issue recently, I wonder if it's worth adding a section to the [Running tasks](https://puppet.com/docs/bolt/latest/bolt_running_tasks.html) page that explains how tasks are uploaded to `/tmp` and how you can configure it if that directory is mounted noexec.

That can definitely be in another ticket if we do want to go that route.

**Comment by hestonhoffman** [comment]
@lucywyman I don't love this solution, but at least it keeps the heading down to a manageable length. Judging from past feedback, some users do not know that the directory has `noexec` applied. So I want to make sure that we're addressing those users. Kind of a chicken and egg scenario

File: documentation/troubleshooting.md (hunk 1 above)
Comment (hestonhoffman) [reply documentation/troubleshooting.md]: Yeah that's true. I think there's a ticket somewhere for improving the tasks docs. I'll add this to that ticket.

**Comment by lucywyman** [comment]
Definitely - this solution seems ideal in that people who *suspect* noexec may be an issue can find it easily, but those who don't will still read it

**Review by lucywyman — APPROVED** [review]

**Review by beechtom — APPROVED** [review]

**Pull request merged**